library/std/src/sys/wasi/thread.rs

```diff
@@ -1,20 +1,116 @@
-#![deny(unsafe_op_in_unsafe_fn)]
-
+use crate::cmp;
 use crate::ffi::CStr;
 use crate::io;
 use crate::mem;
 use crate::num::NonZeroUsize;
+use crate::ptr;
+use crate::sys::os;
 use crate::sys::unsupported;
 use crate::time::Duration;
 
-pub struct Thread(!);
+// #[cfg(all(target_os = "linux", target_env = "gnu"))]
+// use crate::sys::weak::dlsym;
+
+// Add a few symbols not in upstream `libc` just yet.
+mod libc {
+    pub use crate::ffi;
+    pub use crate::mem;
+    pub use libc::*;
+
+    // defined in wasi-libc
+    // https://github.com/WebAssembly/wasi-libc/blob/a6f871343313220b76009827ed0153586361c0d5/libc-top-half/musl/include/alltypes.h.in#L108
+    #[repr(C)]
+    union pthread_attr_union {
+        __i: [ffi::c_int; if mem::size_of::<ffi::c_int>() == 8 { 14 } else { 9 }],
+        __vi: [ffi::c_int; if mem::size_of::<ffi::c_int>() == 8 { 14 } else { 9 }],
+        __s: [ffi::c_ulong; if mem::size_of::<ffi::c_int>() == 8 { 7 } else { 9 }],
+    }
+
+    #[repr(C)]
+    pub struct pthread_attr_t {
+        __u: pthread_attr_union,
+    }
+
+    #[allow(non_camel_case_types)]
+    pub type pthread_t = *mut ffi::c_void;
+
+    extern "C" {
+        // pub fn pthread_self() -> pthread_t;
```

Comment:
Here and below: remove commented-out lines?

Reply:
✅